File: csharp/src/FuelDownloader.Domain/FuelRate.cs
using System.Text.Json.Serialization;

namespace FuelDownloader.Domain
{
    public class FuelRate
    {
        public string ProductCode { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public string AreaCode { get; set; } = string.Empty;
        public string AreaName { get; set; } = string.Empty;
        public DateTime Period { get; set; }
        public decimal Value { get; set; }
        public string Unit { get; set; } = string.Empty;
    }

    // TODO: Add DTOs
    //public class EiaResponse
    //{
    //    public ResponseData Response { get; set; }
    //}

    //public class ResponseData
    //{
    //    public List<FuelData> Data { get; set; }
    //}

    //public class FuelData
    //{
    //    [JsonPropertyName("product")]
    //    public string ProductCode { get; set; }

    //    [JsonPropertyName("product-name")]
    //    public string ProductName { get; set; }

    //    // ... etc
    //}
}

File: go/domain/fuel_rate.go
package domain

import "time"

type FuelRate struct {
	ID          int       `json:"id"`
	Product     string    `json:"product"`
	DuoArea     string    `json:"duoarea"`
	Period      time.Time `json:"period"`
	Value       float64   `json:"value"`
	Units       string    `json:"units"`
	ProductName string    `json:"product_name"`
	AreaName    string    `json:"area_name"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}
